main.go

```diff
@@ -55,17 +65,26 @@ func main() {
 // WatchEvents connects to the k8s API server (using an in-cluster configuration if kubconfig and
 // master are empty) and prints out each event that is observed.
 func WatchEvents(ctx context.Context, master, kubeconfig string) error {
-	config, err := clientcmd.BuildConfigFromFlags(master, kubeconfig)
+	clusterConfig, err := clientcmd.BuildConfigFromFlags(master, kubeconfig)
 	if err != nil {
 		return fmt.Errorf("kubernetes: build config: %w", err)
 	}
 
-	clientset, err := kubernetes.NewForConfig(config)
+	clientset, err := kubernetes.NewForConfig(clusterConfig)
 	if err != nil {
 		return fmt.Errorf("kubernetes: new client: %w", err)
 	}
 
-	lw := cache.NewListWatchFromClient(clientset.CoreV1().RESTClient(), "events", "", fields.Everything())
+	scope := ""
+	if config.Scope == "namespace" {
```

Comment:
This breaks the ability to run out of cluster, which is nice for development (`go run main.go --kubeconfig=$HOME/.kube/config`).  You can do this check, but `config.Scope` should probably just be called `namespace` and it can be passed unmodifed to NewListWatchFromClient.  If you leave it blank, it watches all namespaces.  If you specify a namespace, it uses that namespace.